app/src/test/java/com/codesoom/assignment/application/AuthenticationServiceTest.java

```diff
@@ -0,0 +1,157 @@
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+
+@SpringBootTest
+class AuthenticationServiceTest {
+
+    @Value("${jwt.secret}")
+    private String secret;
+
+    private AuthenticationService authenticationService;
+
+    private final UserRepository userRepository = mock(UserRepository.class);
+
+    private UserLoginDto validUserLoginDto;
+    private UserLoginDto userLoginDtoWithWrongPassword;
+    private UserLoginDto invalidUserLoginDto;
+    private UserLoginDto deletedUserLoginDto;
+    private User validUser;
+    private User deletedUser;
+
+    @BeforeEach
+    void setUp() {
+        JwtUtil jwtUtil = new JwtUtil(secret);
+
+        authenticationService = new AuthenticationService(userRepository, jwtUtil);
+
+        validUser = User.builder()
+                .id(1L)
+                .email("rhfpdk92@naver.com")
+                .password("1234")
+                .name("양승인")
+                .build();
+
+        deletedUser = User.builder()
+                .id(2L)
+                .email("delete@naver.com")
+                .password("password")
+                .name("삭제된유저")
+                .deleted(true)
+                .build();
+
+        validUserLoginDto = UserLoginDto.builder()
+                .email(validUser.getEmail())
+                .password(validUser.getPassword())
+                .build();
+
+        userLoginDtoWithWrongPassword = UserLoginDto.builder()
+                .email(validUser.getEmail())
+                .password("wrong password")
+                .build();
+
+        invalidUserLoginDto = UserLoginDto.builder()
+                .email("doesNotExist@naver.com")
+                .password("password")
+                .build();
+
+        deletedUserLoginDto = UserLoginDto.builder()
+                .email(deletedUser.getEmail())
+                .password(deletedUser.getPassword())
+                .build();
+    }
+
+    @Nested
+    @DisplayName("login() 메소드는")
+    class Describe_login {
+
+        @Nested
+        @DisplayName("유효한 로그인 정보를 주면")
+        class Context_with_valid_login_data {
+
+            @BeforeEach
+            void setUp() {
+                given(userRepository.findByEmail("rhfpdk92@naver.com"))
+                        .willReturn(Optional.of(validUser));
+            }
+
+            @Test
+            @DisplayName("액세스 토큰을 반환한다.")
+            void it_return_access_token() {
+                String accessToken = authenticationService.login(validUserLoginDto);
+                assertThat(accessToken).contains(".");
+            }
+        }
+
+        @Nested
+        @DisplayName("로그인 정보에 해당하는 회원이 없으면")
+        class Context_does_not_exist_user {
+
+            @BeforeEach
+            void setUp() {
+                given(userRepository.findByEmail(invalidUserLoginDto.getEmail()))
+                        .willReturn(Optional.empty());
+            }
+
+            @Test
+            @DisplayName("InvalidUserException을 반환한다.")
```

Comment:
예외는 `return`하지 않습니다. `throw` 하죠.
```suggestion
            @DisplayName("InvalidUserException을 던진다.")
```